src/logic/safe/store/actions/createTransaction.ts

```diff
@@ -184,26 +183,21 @@ export class TxSender {
     )
   }
 
-  async sendTx(): Promise<string> {
-    const { txArgs, isFinalization, from, safeTxHash, txProps, dispatch, txId } = this
+  async sendTx(confirmCallback?: ConfirmEventHandler): Promise<string> {
+    const { txArgs, isFinalization, from, safeTxHash, txProps } = this
 
     const tx = isFinalization ? getExecutionTransaction(txArgs) : getApprovalTransaction(this.safeInstance, safeTxHash)
     const sendParams = createSendParams(from, txProps.ethParameters || {})
-    const promiEvent = tx.send(sendParams)
 
-    // When signing on-chain don't mark as pending as it is never removed
-    if (isFinalization) {
-      // Finalising existing transaction (txId exists)
-      if (txId) {
-        dispatch(addPendingTransaction({ id: txId }))
-      }
-      aboutToExecuteTx.setNonce(txArgs.nonce)
-    }
-
-    return new Promise((resolve, reject) => {
-      promiEvent.once('transactionHash', resolve) // this happens much faster than receipt
-      promiEvent.once('error', reject)
-    })
+    return await tx
+      .send(sendParams)
+      .once('transactionHash', (txHash) => {
+        if (isFinalization) {
+          aboutToExecuteTx.setNonce(txArgs.nonce)
+        }
+        this.onComplete(undefined, confirmCallback, txHash)
+      })
+      .then(({ transactionHash }) => transactionHash)
```

Comment:
Does this have the same behaviour as explicitly throwing inside `.on('error')`?

Reply:
I am not sure. I'll read through the `web3` code later on. I don't think it would harm to re-throw the error there too.

Reply:
Throwing inside `on('error')` actually won't work. It's a different async event. It won't reject the promise, so it won't be caught by the caller.

Reply:
But we could pass `err` to `this.onError` though.

Reply:
The only issue here is that these events persistently exist until the `promiEvent` throws.

Reply:
I just tested it and it seems to be the same behaviour as explicitly throwing. The 50 block error is only visible in the console and is shown as `Transaction still pending. Consider resubmitting with a higher gas price.` to the user.